modules/geo/src/main/java/org/opensearch/geo/search/aggregations/bucket/geogrid/BaseGeoGrid.java

```diff
@@ -69,15 +70,15 @@ protected BaseGeoGrid(String name, int requiredSize, List<BaseGeoGridBucket> buc
         this.buckets = buckets;
     }
 
-    protected abstract Writeable.Reader<B> getBucketReader();
+    protected abstract BaseWriteable.Reader<StreamInput, B> getBucketReader();
 
     /**
      * Read from a stream.
      */
     public BaseGeoGrid(StreamInput in) throws IOException {
         super(in);
         requiredSize = readSize(in);
-        buckets = (List<BaseGeoGridBucket>) in.readList(getBucketReader());
+        buckets = (List<BaseGeoGridBucket>) in.readList((Writeable.Reader<B>) getBucketReader());
```

Comment:
I think you could easily get rid of the casts by using ` BaseWriteable.Reader` in `StreamInput`, this is safe change since the `Reader` subclasses `BaseWriteable.Reader`:

```
public <T> List<T> readList(final BaseWriteable.Reader<StreamInput, T> reader) throws IOException {
        ...
}

private <T, C extends Collection<? super T>> C  readCollection(BaseWriteable.Reader<StreamInput, T> reader, IntFunction<C> constructor, C  empty) {
  ... 
}
```

Reply:
++ I opted against that for this PR because I have a followup PR that does something similar, pulls those methods (e.g., `readList`, `readCollection`) up to `BaseStreamInput` `BaseStreamOutput`.

Reply:
:+1: I am fine with that, I think it would be less work for you if changing that now (just an opinion)